Have to rename libsodium.dll -> sodium.dll so that jnr-ffi could find library

Hi! I downloaded latest release of libsodium (https://download.libsodium.org/libsodium/releases/libsodium-1.0.11-msvc.zip), extracted appropriate `libsodium.dll`, set up `java.library.path` property... but jnr-ffi failed to load sodium library (throws LinkageError). If I rename `libsodium.dll` to `sodium.dll` it works fine. Is it correct behaviour on Windows platform?

OS: Windows 7 SP1 x86
Kalium 0.5.0

Also on windows 10 I had a similar error:
Caused by: java.lang.UnsatisfiedLinkError: unknown
        at jnr.ffi.provider.jffi.NativeLibrary.loadNativeLibraries(NativeLibrary.java:87)

Using procmon I found it was looking for libsodium not in the java library path but in one spot:
C:\ProgramData\Oracle\Java\javapath\sodium.dll

I took libsodium-1.0.11-msvc\x64\Release\v140\dynamic\libsodium.dll copied and renamed it to the path and now the play-kalium example runs :)

--- a/README.md
+++ b/README.md
@@ -34,6 +34,4 @@ Add as a Maven dependency at your project:
 kalium is a work in progress, feedback, bug reports and patches are always welcome.
 
 [![Build Status](https://travis-ci.org/abstractj/kalium.png?branch=master)](https://travis-ci.org/abstractj/kalium)
-
-
-
+[![Build status](https://ci.appveyor.com/api/projects/status/github/abstractj/kalium?branch=master&svg=true)](https://ci.appveyor.com/project/abstractj/kalium/branch/master)

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,0 +1,22 @@
+version: '{build}'
+os: Windows Server 2012
+install:
+  - cmd: SET JAVA_HOME=C:\Program Files (x86)\Java\jdk1.8.0
+  - cmd: SET LIBSODIUM_VERSION=1.0.3
+  - ps: |
+      Add-Type -AssemblyName System.IO.Compression.FileSystem
+      if (!(Test-Path -Path "C:\libsodium" )) {
+        (New-Object System.Net.WebClient).DownloadFile(
+          "https://github.com/jedisct1/libsodium/archive/$env:LIBSODIUM_VERSION.zip",
+          'C:\libsodium.zip'
+        )
+        [System.IO.Compression.ZipFile]::ExtractToDirectory("C:\libsodium.zip", "C:\")
+        Rename-Item C:\libsodium-$env:LIBSODIUM_VERSION C:\libsodium
+      }
+  - cmd: msbuild C:\libsodium\libsodium.vcxproj /property:Configuration=ReleaseDLL /verbosity:minimal /logger:"C:\Program Files\AppVeyor\BuildAgent\Appveyor.MSBuildLogger.dll"
+  - cmd: SET PATH=C:\libsodium\Build\ReleaseDLL\Win32;%PATH%
+build_script:
+  - mvn clean install -q --batch-mode -DargLine=-Dfile.encoding=UTF-8
+cache:
+  - C:\libsodium
+  - C:\Users\appveyor\.m2

--- a/src/main/java/org/abstractj/kalium/NaCl.java
+++ b/src/main/java/org/abstractj/kalium/NaCl.java
@@ -17,6 +17,7 @@
 package org.abstractj.kalium;
 
 import jnr.ffi.LibraryLoader;
+import jnr.ffi.Platform;
 import jnr.ffi.annotations.In;
 import jnr.ffi.annotations.Out;
 import jnr.ffi.byref.LongLongByReference;
@@ -30,7 +31,16 @@ public class NaCl {
         return sodium;
     }
 
-    private static final String LIBRARY_NAME = "sodium";
+    private static final String LIBRARY_NAME = libraryName();
+
+    private static String libraryName() {
+        switch (Platform.getNativePlatform().getOS()) {
+            case WINDOWS:
+                return "libsodium";
+            default:
+                return "sodium";
+        }
+    }
 
     private static final class SingletonHolder {
         public static final Sodium SODIUM_INSTANCE =

